dkpro-core-io-nyt-asl/src/main/java/de/tudarmstadt/ukp/dkpro/core/io/nyt/NYTCollectionReader.java

```diff
@@ -0,0 +1,171 @@
+package de.tudarmstadt.ukp.dkpro.core.io.nyt;
+
+import java.io.IOException;
+import java.net.URL;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+
+import javax.xml.parsers.ParserConfigurationException;
+
+import org.apache.uima.UimaContext;
+import org.apache.uima.cas.CAS;
+import org.apache.uima.cas.CASException;
+import org.apache.uima.collection.CollectionException;
+import org.apache.uima.fit.component.CasCollectionReader_ImplBase;
+import org.apache.uima.fit.descriptor.ConfigurationParameter;
+import org.apache.uima.jcas.JCas;
+import org.apache.uima.jcas.cas.StringArray;
+import org.apache.uima.resource.ResourceInitializationException;
+import org.apache.uima.util.Level;
+import org.apache.uima.util.Logger;
+import org.apache.uima.util.Progress;
+import org.apache.uima.util.ProgressImpl;
+
+import com.nytlabs.corpus.NYTCorpusDocument;
+
+import de.tudarmstadt.ukp.dkpro.io.nyt.metadata.NYTArticleMetaData;
+import de.tudarmstadt.ukp.dkpro.core.api.metadata.type.DocumentMetaData;
+
+public class NYTCollectionReader extends CasCollectionReader_ImplBase {
```

Comment:
It might not be obvious from the test data, but I would need to iterate over all XML files contained in multiple .tgz archives in multiple directories. I am currently trying to use `nextFile()` to get the archive files and then iterate over the XML files using my existing iterator(s). It feels a bit cumbersome, tough, also because of necessary conversions between Path and Resource (but that maybe just means that it makes sense to stick to Resource).  Could you please advise on what the best practice looks like in this case?